base/database/testing.go

```diff
@@ -152,6 +152,16 @@ func CheckAdvisoriesAccountData(t *testing.T, rhAccountID int, advisoryIDs []int
 	assert.Equal(t, systemsAffected*len(advisoryIDs), sum, "sum of systems_affected does not match")
 }
 
+func CheckSystemUpdatesCount(t *testing.T, systemID int) []int {
+	var cnt []int
+	assert.NoError(t, Db.Table("system_package spkg").
+		Select("json_array_length(spkg.update_data::json) as len").
+		Where("spkg.update_data is not null").
+		Where("spkg.system_id = ? ", systemID).
```

Comment:
add `spkg.rh_account_id = ?` here otherwise it will scan all system_package partitions.